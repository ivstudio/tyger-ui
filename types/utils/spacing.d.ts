export declare const spacing: {
    readonly '0': "0px";
    readonly '2': "2px";
    readonly '4': "4px";
    readonly '8': "8px";
    readonly '12': "12px";
    readonly '16': "16px";
    readonly '24': "24px";
    readonly '32': "32px";
    readonly '40': "40px";
    readonly '48': "48px";
    readonly '64': "64px";
    readonly '80': "80px";
    readonly '96': "96px";
    readonly '160': "160px";
};
export declare type TSpacing = keyof typeof spacing;
