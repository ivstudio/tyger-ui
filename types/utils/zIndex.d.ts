export declare const zIndex: {
    readonly mobileStepper: 1000;
    readonly appBar: 1100;
    readonly drawer: 1200;
    readonly modal: 1300;
    readonly snackbar: 1400;
    readonly tooltip: 1500;
};
export declare type TZIndex = keyof typeof zIndex;
