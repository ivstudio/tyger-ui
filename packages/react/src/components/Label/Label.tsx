'use client';
import { forwardRef } from 'react';
import styled, { css } from 'styled-components';

type LabelProps = {
    required?: boolean;
    disabled?: boolean;
    htmlFor?: string;
    children: React.ReactNode | React.ReactNode[];
};

// Define the LabelRoot styled component
const LabelRoot = styled.label<LabelProps>`
    /* Set base styles for all labels */
    display: block;
    font-size: ${({ theme: { fontSize } }) => `${fontSize[-1]}`};
    margin: ${({ theme: { spacing } }) => `0 0 ${spacing[4]}`};

    /* Add asterisk to required labels */
    &::after {
        content: ${({ required }) => (required ? "'*'" : "''")};
        padding-left: ${({ theme: { spacing } }) => spacing[4]};
    }

    /* Set styles for disabled labels */
    ${({ disabled, theme: { input } }) => {
        if (disabled) {
            return css`
                color: ${input.disabled.color};
            `;
        }
    }}
`;

// Define the Label component
const Label = forwardRef<HTMLLabelElement, LabelProps>(
    (props, forwardedRef) => {
        const { required, disabled, htmlFor, children, ...baseProps } = props;
        return (
            <LabelRoot
                {...baseProps}
                ref={forwardedRef}
                htmlFor={htmlFor}
                required={required}
                disabled={disabled}
            >
                {children}
            </LabelRoot>
        );
    }
);

Label.displayName = 'Label';
export default Label;
