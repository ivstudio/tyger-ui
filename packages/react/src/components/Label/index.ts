'use client';
export { default } from './Label';
