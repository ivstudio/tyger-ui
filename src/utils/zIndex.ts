export const zindex = {
    mobileStepper: 1000,
    appBar: 1100,
    drawer: 1200,
    modal: 1300,
    snackbar: 1400,
    tooltip: 1500,
};
export type TZIndex = keyof typeof zindex;
