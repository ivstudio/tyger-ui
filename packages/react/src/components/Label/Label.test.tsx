import React from 'react';

import { render } from '../../../test/utils/mockProvider';
import Label from './Label';

describe('Label', () => {
    it('renders correctly', () => {
        const { container } = render(
            <Label required htmlFor="firstName">
                Jon Doe
            </Label>
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders label with asterisk when required prop is true', () => {
        const { getByText } = render(
            <Label required htmlFor="firstName">
                Jon Doe
            </Label>
        );
        const asteriskElement = getByText(/Jon Doe */i).closest('label');
        expect(asteriskElement).toBeInTheDocument();
    });

    it('applies disabled styles when disabled prop is true', () => {
        const { getByText } = render(
            <Label disabled htmlFor="firstName">
                Jon Doe
            </Label>
        );

        const labelElement = getByText(/Jon Doe/i).closest('label');
        expect(labelElement).toHaveAttribute('disabled');
    });
});
