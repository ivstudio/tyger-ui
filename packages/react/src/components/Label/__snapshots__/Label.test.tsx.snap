// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly 1`] = `
<label
  class="sc-aXZVg izzKfH"
  for="firstName"
  required=""
>
  Jon Doe
</label>
`;
